File: src/main/java/net/shadowfacts/underwaterutilities/util/energy/RFAdapter.java
package net.shadowfacts.underwaterutilities.util.energy;

import cofh.api.energy.IEnergyStorage;
import net.darkhax.tesla.api.implementation.BaseTeslaContainer;

/**
 * @author shadowfacts
 */
public interface RFAdapter extends IEnergyStorage {

	BaseTeslaContainer getTeslaContainer();

	@Override
	default int receiveEnergy(int maxReceive, boolean simulate) {
		return (int)getTeslaContainer().givePower(maxReceive, simulate);
	}

	@Override
	default int extractEnergy(int maxExtract, boolean simulate) {
		return (int)getTeslaContainer().takePower(maxExtract, simulate);
	}

	@Override
	default int getEnergyStored() {
		return (int)getTeslaContainer().getStoredPower();
	}

	@Override
	default int getMaxEnergyStored() {
		return (int)getTeslaContainer().getCapacity();
	}

}

File: src/main/kotlin/net/shadowfacts/underwaterutilities/util/energy/RFAdapter.kt
package net.shadowfacts.underwaterutilities.util.energy

import cofh.redstoneflux.api.IEnergyStorage
import net.darkhax.tesla.api.implementation.BaseTeslaContainer

/**
 * @author shadowfacts
 */
interface RFAdapter: IEnergyStorage {

	val teslaContainer: BaseTeslaContainer

	override fun receiveEnergy(maxReceive: Int, simulate: Boolean): Int {
		return teslaContainer.givePower(maxReceive.toLong(), simulate).toInt()
	}

	override fun extractEnergy(maxExtract: Int, simulate: Boolean): Int {
		return teslaContainer.takePower(maxExtract.toLong(), simulate).toInt()
	}

	override fun getEnergyStored() = teslaContainer.storedPower.toInt()

	override fun getMaxEnergyStored() = teslaContainer.capacity.toInt()

}
